### PR TITLE
imagebuildah: honor build output even if build container is not `commited`

### PR DESCRIPTION
**Pull request by flouthoc**
For cases like where Containerfile only contains `FROM alpine` or `FROM
<imagename>` buildah will just perform name tagging and will not
actually commit contents of build container since there is actually no
changes made in the build process. However previously we only used to
export build output when we were actually commiting contents so `-o` or
`--output` was ignored for such edge-cases following commit ensures that
we honor `-o` or `--output` in all scenarios.

Example: Should be functional now
```Dockerfile
FROM alpine
```
```console
buildah build -t test -o type=tar,dest=out.tar .
```

Closes: https://github.com/containers/buildah/issues/4094

#### Does this PR introduce a user-facing change?

```release-note
fix: users can generate build output of image even when contents of build container is not commited
```

### CODE REVIEW

**Comment by flouthoc** [comment]
@rhatdan Rebased @containers/buildah-maintainers @giuseppe @vrothberg PTAL

File: imagebuildah/stage_executor.go
Comment (rhatdan) [inline imagebuildah/stage_executor.go:1745]: if err := ...

**Review by rhatdan — APPROVED** [review]

File: imagebuildah/stage_executor.go
Comment (flouthoc) [reply imagebuildah/stage_executor.go]: fixed

**Comment by flouthoc** [comment]
@rhatdan Could PTAL again.

**Comment by rhatdan** [comment]
LGTM
@giuseppe @vrothberg @umohnani8 PTAL

**Review by vrothberg — APPROVED** [review]
/lgtm
/hold

**Pull request merged**
